### DISCUSSION TITLE
v2 - wrong modules types being output from npm run dev

### DISCUSSION THREAD

**Discussion by mist998** [medusajs/medusa#10039] [Q&A]
The linked models didn't get outputting every time we run `npm run dev`

even if we could add the line like `customers?: Maybe<Array<Maybe<Customer>>> ` , but it would get rewrite/remove by `npm run dev` immediately

then the `npm run build` would hit ts errors

![image](https://github.com/user-attachments/assets/d65532f2-4dfb-4121-9799-c4f70d7e7822)

![image](https://github.com/user-attachments/assets/beac85c5-2158-481b-8d7f-ccd34c55a1b8)

the package.json
```
   ...,
  "dependencies": {
    "@medusajs/admin-sdk": "rc",
    "@medusajs/framework": "rc",
    "@medusajs/medusa": "rc",
    "@medusajs/medusa-cli": "rc",
    "@mikro-orm/core": "5.9.7",
    "@mikro-orm/knex": "5.9.7",
    "@mikro-orm/migrations": "5.9.7",
    "@mikro-orm/postgresql": "5.9.7",
    "awilix": "^8.0.1",
    "cors": "^2.8.5",
    "medusa-core-utils": "^1.2.3",
    "pg": "^8.13.0",
    "typeorm": "^0.3.20"
  },
```

how to workaround this issue?

**Comment by carlos-r-l-rodrigues** [comment]
Hi, could you please share your DML entities and link definition?

**Comment by carlos-r-l-rodrigues** [comment]
I also noticed you are using the `rc` version. Could you please update to the `latest` version, where some issues related to entity generation were fixed.

**Comment by carlos-r-l-rodrigues** [comment]
Please feel free to reopen the issue in case the new releases haven't fixed this.
Thank you.
